src/content/developers/docs/smart-contracts/upgrading-smart-contracts/index.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Upgrading smart contracts 
+description: An overview of upgrade patterns for Ethereum smart contracts
+sidebar: true
+---
+
+# Upgrading Smart Contracts {#upgrading-smart-contracts}
+
+Smart contracts on Ethereum are self-executing programs that run in the Ethereum Virtual Machine (EVM). These programs are immutable by design, which prevents any updates to the business logic once the contract is deployed.  
+
+While immutability is necessary for trustlessness, decentralization, and security of smart contracts, it may be a drawback in certain cases. For instance, immutable code can make it impossible for developers to fix vulnerable contracts. 
+
+However, increased research into improving smart contracts has led to the introduction of several upgrade patterns. These upgrade patterns enable developers to upgrade smart contracts (while preserving immutability) by placing business logic in different contracts. 
+
+## Prerequisites {#prerequisites}
+
+You should have a good understanding of [smart contracts](/developers/docs/smart-contracts/), [smart contract anatomy](/developers/docs/smart-contracts/anatomy/), and the [Ethereum Virtual Machine (EVM)](/developers/docs/evm/). This guide also assumes readers have a grasp of programming in Solidity. 
```

Comment:
Is this information actually specific to Solidity?
```suggestion
You should have a good understanding of [smart contracts](/developers/docs/smart-contracts/), [smart contract anatomy](/developers/docs/smart-contracts/anatomy/), and the [Ethereum Virtual Machine (EVM)](/developers/docs/evm/). This guide also assumes readers have a grasp of programming smart contracts. 
```

Reply:
I put that there since the page mentions things like function selectors (which link to the Solidity documentation).